### PR TITLE
Add Security and Privay section

### PR DESCRIPTION
**Pull request by silviapfeiffer**
As proposed by David in https://github.com/w3c/webvtt/issues/323 .

There's also some cleanup using format.py

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -5751,6 +5779,43 @@ and registration with IANA.</p>
...
+
+<p>As with any text-based format, it is possible to construct malicious content that might cause
+buffer over-runs, value overflows (e.g. string representations of integers that overflow a given
+word length), and the like. Implementers should take care in implementing a parser that over-long
+lines, field values, or encoded values do not cause security problems.</p>
+
+<h3 id=styling-security>Styling-related security</h3>
+
+<p>WebVTT can embed style ‘snippets’, and they in turn can cause the loading of external style
+sheets through the use of a CSS “@import” rule, in user-agents that support CSS. Under these
```
Comment (zcorpan) [inline index.bs]: STYLE blocks can't load any external resources; in fact it is important for privacy that they cannot. This section should discuss that.

File: index.bs [hunk 2]
```diff
@@ -5751,6 +5779,43 @@ and registration with IANA.</p>
...
+circumstances, the security and privacy considerations of CSS apply. In addition, it is possible for
+a user-agent to offer user style-sheets, and their presence and nature might be detectable by
+scripts running in the same user-agent (e.g. browser). This might enable fingerprinting of the user
```
Comment (zcorpan) [inline index.bs]: I don't think it is detectable, actually. The `getComputedStyle` API only supports the `::before` and `::after` pseudo-elements, not `::cue`.

File: index.bs [hunk 3]
```diff
@@ -5751,6 +5779,43 @@ and registration with IANA.</p>
...
+file that is designed not to present captions or subtitles, but instead to provide timed input
+(‘triggers’) to a script system. A poorly-written script or script system might then cause
+security, privacy or other problems; however, this consideration really applies to the script
+system. Since WebVTT supplies these triggers at their timestamps, a malicious file might present
+such triggers very rapidly, perhaps causing undue resource consumption.</p>
+
+<h3 id=privacy-of-preference>Privacy of preference</h3>
+
+<p>A user-agent that selects, and causes to download or interpret a WebVTT file, might indicate to
+the origin server that the user has a need for captions or subtitles. That is a (small) piece of
```
Comment (zcorpan) [inline index.bs]: Also language preference of the user.
https://html.spec.whatwg.org/#honor-user-preferences-for-automatic-text-track-selection

File: index.bs [hunk 4]
```diff
@@ -5751,6 +5779,43 @@ and registration with IANA.</p>
...
+sheets through the use of a CSS “@import” rule, in user-agents that support CSS. Under these
+circumstances, the security and privacy considerations of CSS apply. In addition, it is possible for
+a user-agent to offer user style-sheets, and their presence and nature might be detectable by
+scripts running in the same user-agent (e.g. browser). This might enable fingerprinting of the user
+or reveal aspects of the user’s preferences (e.g. the choice of a large font size and/or
+high-contrast colors might indicate a user with visual impairments).</p>
+
+<h3 id=scripting-security>Scripting-related security</h3>
+
+<p>WebVTT does not include or enable scripting. However, it is possible to construct and deliver a
```
Comment (zcorpan) [inline index.bs]: Could also stress that it is important that user agents do not support a way to execute script embedded in a WebVTT file.

**Comment by silviapfeiffer** [comment]
All great points, please do! I was just trying to help a little.

**Comment by zcorpan** [comment]
I cleaned up this branch by force-pushing a new branch based on `gh-pages` and cherry-picking 49604f508a3f5a490227c43ab30a57c4305272d2 and 1cb3b75ba967b4912e7d3bc9cab19fd17ac94eeb

**Comment by dwsinger** [comment]
I'd be really happy for Simon (or others) to improve and/or correct this section. Privacy and security considerations generally are best when multiple people think about how privacy or security might be compromised.

**Comment by zcorpan** [comment]
Marked as non-substantive for IPR from ash-nazg.

**Comment by zcorpan** [comment]
Ready for review. (I will squash and rebase and rerun bikeshed to resolve the conflict before merging.)

File: index.bs [hunk 5]
```diff
@@ -5751,6 +5780,55 @@ and registration with IANA.</p>
...
+word length), and the like. Implementers should take care in implementing a parser that over-long
+lines, field values, or encoded values do not cause security problems.</p>
+
+<h3 id=styling-security>Styling-related privacy and security</h3>
+
+<p>WebVTT can embed CSS style sheets, which will be applied in user agents that support CSS. Under
+these circumstances, the privacy and security considerations of CSS apply, with the following
+caveats.</p>
+
+<p>However, such style sheets <a href="#style-no-external-resources">cannot fetch any external
```
Comment (silviapfeiffer) [inline index.bs]: The"However" sounds weird here. Maybe remove?

Comment (zcorpan) [inline index.bs]: Done

**Comment by dwsinger** [comment]
Thank you all! LGTM

**Comment by silviapfeiffer** [comment]
Thanks for taking it over, Simon!

**Pull request merged**